refactor(Stock Registry)

lomamech opened the pull request:
Make stock registry link Search Readable
- in Stock Inventory, Stock lots and Stock Registry
- Implement component bh-flux-select for Inventory Reason
- Implement test E2E for bh-flux-select

this issue reference #2139 and #2130

jniles commented on client/src/modules/stock/inventories/modals/search.modal.js:
```diff
@@ -60,10 +73,27 @@ function SearchInventoriesModalController(data, Notify, Instance, Store, Periods
 
   vm.submit = function submit() {
+    // Set the label of status Value
+    if (vm.searchQueries.status) {
```
I can imagine that other developers will want to use this too.

Why don't you make an object map and put it in a service (like the `StockService`)?  That way, you could do this:
```js
// this is in a stock service
service.statusLabelMap = {
  'sold_out' : 'STOCK.STATUS.SOLD_OUT',
  // ... etc ...
};

// this is in your submit() function.
if (vm.searchQueries.status) {
  displayValues.status = service.statusLabelMap[vm.searchQueries.status];
}
```
That way, if multiple controllers need to have the status --> i18n key mapping, they can just import your service!

lomamech replied:
Sorry for the few Comment, some times when I update my branch after resolving a conflict, I lose the message of my former Commit

jniles commented on client/src/js/components/bhFluxSelect.js:
```diff
@@ -0,0 +1,50 @@
+    controller  : FluxSelectController,
+    bindings    : {
+      onChange : '&',
+      fluxId : '<?',
```
It looks like this is a `ui-select multiple`.  Should this name be `fluxIds` then (in plural)?  Since it could take multiple flux ids?

jniles reviewed (commented):
Just a few comments here.

lomamech committed:
refactor(Stock Inventory Registry)
- Update bhFluxSelect fluxIds instead fluxId
- Add function in stock Service who return Status instead a big
  Alternative Structure

lomamech committed:
Resolve conflict with Branch Master

jniles commented on client/src/js/services/StockService.js:
```diff
@@ -219,6 +219,20 @@ function StockService(Api, Filters, AppCache, Periods, $httpParamSerializer, Lan
   }
 
+    /** Get label for purchase Status */
+  function statusLabelMap(status) {
```
:+1:

jniles commented on client/src/modules/stock/inventories/modals/search.modal.js:
```diff
@@ -60,10 +73,17 @@ function SearchInventoriesModalController(data, Notify, Instance, Store, Periods
   vm.submit = function submit() {
+    // Set the label of status Value
+    if (vm.searchQueries.status) {
+      displayValues.status = Stock.statusLabelMap(vm.searchQueries.status);
```
Beautiful!

jniles reviewed (approved):
LGTM!

bors r+
